internal/bminventory/inventory.go

```diff
@@ -3760,7 +3757,38 @@ func computeProxyHash(proxy *models.Proxy) (string, error) {
 	return fmt.Sprintf("%x", bs), nil
 }
 
-func validateProxySettings(httpProxy, httpsProxy, noProxy, ocpVersion *string) error {
+func validateProxySettings(params interface{}, ocpVersion *string) error {
+	var (
+		httpProxy       *string
+		httpsProxy      *string
+		noProxy         *string
+		validateVersion bool
+	)
+	switch p := params.(type) {
+	case models.ClusterCreateParams:
+		httpProxy = p.HTTPProxy
+		httpsProxy = p.HTTPSProxy
+		noProxy = p.NoProxy
+		validateVersion = true
+	case models.V2ClusterUpdateParams:
+		httpProxy = p.HTTPProxy
+		httpsProxy = p.HTTPSProxy
+		noProxy = p.NoProxy
+		validateVersion = true
+	case models.InfraEnvCreateParams:
+		httpProxy = p.Proxy.HTTPProxy
+		httpsProxy = p.Proxy.HTTPSProxy
+		noProxy = p.Proxy.NoProxy
+		validateVersion = false
+	case models.InfraEnvUpdateParams:
+		httpProxy = p.Proxy.HTTPProxy
+		httpsProxy = p.Proxy.HTTPSProxy
+		noProxy = p.Proxy.NoProxy
+		validateVersion = false
+	default:
+		return errors.New("internal error")
```

Comment:
Consider better error message, e.g.
```suggestion
		return fmt.Errorf("failed to validate proxy setting due to unexpected interface type: %s", params.(type))
```

Reply:
will do.